### ISSUE TITLE
Incorrect highlighting in nested use statements

### ISSUE THREAD

**Issue by afetisov** [intellij-rust/intellij-rust#7508]
## Environment

* **IntelliJ Rust plugin version:** 0.4.151.3973-211-nightly
* **Rust toolchain version:** 1.53.0 (53cb7b09b 2021-06-17) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.1.2 (CL-211.7442.42)
* **Operating system:** Linux 5.4.0-77-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

In nested use statements only the first unused import is highlighted. If it is removed, the next one will be highlighted, and so on until no unused imports remain. If the statements are flattened, then all unused imports are correctly highlighted.

## Steps to reproduce

```rust
pub mod bar {
    pub mod foo {
        pub fn fst() {}

        pub fn snd() {}
    }
    
    pub mod foo2 {
        pub fn fst2() {}

        pub fn snd2() {}
    }
}

use bar::{
    foo::{fst, snd},
       // ^^^ weak warning
    foo2::{fst2, snd2},
};
```

Weak warning: "unused imports: `fst2`, `fst`, `snd2`, `snd` Help: remove the whole `use` item"

**Comment by Kobzol** [comment]
Oh, first I thought that this is highlighted by the new experimental unused imports inspection, but this is coming from `cargo check` - you have enabled on-the-fly linter analysis, right?

@mchernyavsky I checked the output of `cargo check` in this case and it returns all of the sub-spans of `fst`, `snd`, `fst2` and `snd2`. However, `RsExternalLinterFilteredMessage::filterMessage` only takes the first span. I posted a naive fix [here](), but I'm not sure if it's valid.
